v2/lints/cabf_br/lint_ocsp_id_pkix_ocsp_nocheck_ext_not_included.go

```diff
@@ -0,0 +1,64 @@
+/*
+ * ZLint Copyright 2020 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package cabf_br
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/v2/lint"
+	"github.com/zmap/zlint/v2/util"
+)
+
+type OCSPIDPKIXOCSPNocheckExtNotIncluded struct{}
+
+func (l *OCSPIDPKIXOCSPNocheckExtNotIncluded) Initialize() error {
+	return nil
+}
+
+func (l *OCSPIDPKIXOCSPNocheckExtNotIncluded) CheckApplies(c *x509.Certificate) bool {
+	if util.HasEKU(c, x509.ExtKeyUsageOcspSigning) {
+		return true
+	}
+	return false
+}
+
+func (l *OCSPIDPKIXOCSPNocheckExtNotIncluded) Execute(c *x509.Certificate) *lint.LintResult {
+	if !util.HasEKU(c, x509.ExtKeyUsageOcspSigning) {
+		// This point should never be reached, because this LINT only applies to certificates with this EKU
+		return &lint.LintResult{Status: lint.Fatal}
+	}
```

Comment:
Do we need to duplicate this? It seems like we shouldn't, but if there's some reason I've overlooked, it also seems like

```
if !CheckApplies(c) {
```

would be better. But I don't think it's needed :)

Reply:
Agree, I think that this can be dropped.

Reply:
I updated the code with the proposal by @sleevi . I would like to let this check in here, even if it is redundant because it is defensive programming and one should never make assumptions on the value fed into a function as long as it not validated in the method itself.

Reply:
It’s actually possible to be too defensive. I totally hear you about not wanting to make assumptions, but as engineers, we also write code with defined or guaranteed behaviors, according to interface contracts.

The reason why I wasn’t sure, but Zakir confirmed my memory was correct, is that the current program defines it an invalid sequence to call `Execute` without calling `CheckApplies`. I totally understand not wanting to assume, but it’s a bit like how the code guarantees `Initialize` will be called beforehand, or like how it’s “undefined” (or “invalid” or “wrong”, depending on language) to use memory before it’s been initialized, or to write to a file handle you haven’t opened. Some things it’s OK to take for granted, because that’s the defined contract, and if anything crashes, it’s because the calling code is wrong, not this code.

I ultimately defer to @zakird and @cpu how strongly they feel here, because I do seem to recall there might be a few of the newer lints not consistently following this, but I think we can get away without this, saying it’s “programmer error” to call if `CheckApplies` returns false.

Reply:
I think there is a significant difference between relying on the compiler / runtime environment to correctly initialize the memory or call the constructor of an object and on relying on the programmer to call two methods in the correct order. But in the end, I'm following the advice of the maintainer :-)